fvm/environment/programs.go

```diff
@@ -35,6 +35,10 @@ type Programs struct {
 	accounts Accounts
 
 	transactionPrograms TransactionPrograms
+
+	// NOTE: non-address programs are not reusable across transactions, hence
+	// they are kept out of the derived data database.
```

Comment:
This comments makes me wonder what is the "derived data". Is that the `TransactionPrograms`?

Could you add comments to other fields to indicate what are derived, what are not.

Reply:
TransactionPrograms is renamed to DerivedTransactionData in #3516.  The plan is to cache all pre-compute / derived data (e.g., programs, meter settings, protocol version, etc) in this.   Everything in this package will be derived from ledger data.